### PR TITLE
More doc comments for commonly used APIs

### PR DESCRIPTION
**Pull request by rowanmiller**
This gets doc comments onto the bulk of APIs that would be used in most applications.

### CODE REVIEW

File: src/Microsoft.EntityFrameworkCore.InMemory/Extensions/InMemoryServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -15,8 +15,40 @@
...
+        /// <summary>
+        ///     <para>
+        ///         Adds the services required by the in-memory database provider for Entity Framework
+        ///         to an <see cref="IServiceCollection" />. You use this method when using dependency injection
+        ///         in your application, such as with ASP.NET. For more information on setting up dependency
+        ///         injection, see http://go.microsoft.com/fwlink/?LinkId=526890.
+        ///     </para>
+        ///     <para>
+        ///         You only need to use this functionality when you want Entity Framework to resolve the services it uses
+        ///         from an external <see cref="IServiceCollection" />. If you are not using an external
```
Comment (divega) [inline src/Microsoft.EntityFrameworkCore.InMemory/Extensions/InMemoryServiceCollectionExtensions.cs]: Not sure saying we resolve from an external `IServiceCollection` is clear. I would use `IServiceProvider` here and in the next instance in the same paragraph as well.

Comment (divega) [inline src/Microsoft.EntityFrameworkCore.InMemory/Extensions/InMemoryServiceCollectionExtensions.cs]: or maybe just "dependency injection container".

Comment (rowanmiller) [inline src/Microsoft.EntityFrameworkCore.InMemory/Extensions/InMemoryServiceCollectionExtensions.cs]: > or maybe just "dependency injection container".

👍

File: src/Microsoft.EntityFrameworkCore.InMemory/Infrastructure/InMemoryEventId.cs [hunk 2]
```diff
@@ -1,11 +1,23 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using Microsoft.Extensions.Logging;
+
 namespace Microsoft.EntityFrameworkCore.Infrastructure
 {
+    /// <summary>
+    ///     Values that are used as the eventId when logging messages from this assembly via <see cref="ILogger" />.
```
Comment (divega) [inline src/Microsoft.EntityFrameworkCore.InMemory/Infrastructure/InMemoryEventId.cs]: Technically the fact that are messages from this "assembly" my matter and be correct, but it seems a bit obscure for the user. Can we use something else? E.g. "provider"?

File: src/Microsoft.EntityFrameworkCore.Relational/Infrastructure/RelationalEventId.cs [hunk 3]
```diff
@@ -1,19 +1,64 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using Microsoft.Extensions.Logging;
+
 namespace Microsoft.EntityFrameworkCore.Infrastructure
 {
+    /// <summary>
+    ///     Values that are used as the eventId when logging messages from this assembly via <see cref="ILogger" />.
```
Comment (divega) [inline src/Microsoft.EntityFrameworkCore.Relational/Infrastructure/RelationalEventId.cs]: Idem re "assembly" maybe from "Entity Framework Core" this time?

File: src/Microsoft.EntityFrameworkCore.Sqlite/Infrastructure/SqliteServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -20,8 +20,42 @@
...
+        /// <summary>
+        ///     <para>
+        ///         Adds the services required by the SQLite database provider for Entity Framework
+        ///         to an <see cref="IServiceCollection" />. You use this method when using dependency injection
+        ///         in your application, such as with ASP.NET. For more information on setting up dependency
+        ///         injection, see http://go.microsoft.com/fwlink/?LinkId=526890.
+        ///     </para>
+        ///     <para>
+        ///         You only need to use this functionality when you want Entity Framework to resolve the services it uses
+        ///         from an external <see cref="IServiceCollection" />. If you are not using an external
```
Comment (divega) [inline src/Microsoft.EntityFrameworkCore.Sqlite/Infrastructure/SqliteServiceCollectionExtensions.cs]: Same comment as before re service collection vs. provider/container. Applies to both instances in this paragraph.

**Comment by divega** [comment]
I made a few suggestions but it looks very good. 
:shipit:

File: src/Microsoft.EntityFrameworkCore.Relational/Infrastructure/RelationalEventId.cs (hunk 3 above)
Comment (rowanmiller) [inline src/Microsoft.EntityFrameworkCore.Relational/Infrastructure/RelationalEventId.cs]: I went with `...when logging messages from a relational database provider`

**Pull request merged**